### PR TITLE
Internal: Dynamic tag promotion doesn't appear when the license is inactive or expired [ED-5143]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js [hunk 1]
```diff
@@ -13,10 +13,30 @@ module.exports = Marionette.Behavior.extend( {
 		}
 	},
 
+	shouldRenderTools() {
+		const hasDefault = this.getOption( 'dynamicSettings' ).default,
```
Comment (StyleShit) [inline assets/dev/js/editor/components/dynamic-tags/control-behavior.js:17]: I have no idea why, I just kept the original behavior

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js [hunk 2]
```diff
@@ -13,10 +13,30 @@ module.exports = Marionette.Behavior.extend( {
...
+			hasPro = elementor.helpers.hasPro(),
+			hasTags = this.getOption( 'tags' ).length > 0,
+			shouldHavePromotion = ! hasPro || elementor.helpers.hasProAndNotConnected(),
+			shouldHaveSwitcher = hasPro && hasTags;
+
+		if ( hasDefault ) {
+			return false;
+		}
```
Comment (StyleShit) [inline assets/dev/js/editor/components/dynamic-tags/control-behavior.js:25]: I have no idea why, I just kept the original behavior

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js [hunk 3]
```diff
@@ -13,10 +13,30 @@ module.exports = Marionette.Behavior.extend( {
 		}
 	},
 
+	shouldRenderTools() {
+		const hasDefault = this.getOption( 'dynamicSettings' ).default,
+			hasPro = elementor.helpers.hasPro(),
+			hasTags = this.getOption( 'tags' ).length > 0,
+			shouldHavePromotion = ! hasPro || elementor.helpers.hasProAndNotConnected(),
+			shouldHaveSwitcher = hasPro && hasTags;
```
Comment (Nevoss) [inline assets/dev/js/editor/components/dynamic-tags/control-behavior.js:21]:
```suggestion
		const hasDefault = this.getOption( 'dynamicSettings' ).default,
			isFeatureAvalibleToUser = elementor.helpers.hasPro() && ! elementor.helpers.hasProAndNotConnected(),
		   	hasTags = this.getOption( 'tags' ).length > 0;
		   	
		   	if (hasDefault) {
		   	 	return false;
		   	} 
		   	
		   	return ! isFeatureAvalibleToUser || hasTags
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js (hunk 3 above)
Comment (StyleShit) [reply assets/dev/js/editor/components/dynamic-tags/control-behavior.js]: Done. 
And yes, the calculation shouldn't be here for sure... The whole module shouldn't be here...
But I didn't want to have dependency and didn't want to change the code as much.

Comment (StyleShit) [reply assets/dev/js/editor/components/dynamic-tags/control-behavior.js]: Done. 
And yes, the calculation shouldn't be here for sure... The whole module shouldn't be here...
But I didn't want to have dependency and didn't want to change the code as much.

**Review by Nevoss — DISMISSED** [review]

File: tests/jest/unit/assets/dev/js/editor/components/dynamic-tags/control-behavior.test.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+			expected: true,
+			description: 'Should render tools when there is no Pro',
+		},
+		{
+			hasDefault: true,
+			hasPro: false,
+			isProConnected: false,
+			hasTags: false,
+			expected: false,
+			description: 'Should no render tools when there is no Pro and there is default',
```
Comment (matipojo) [inline tests/jest/unit/assets/dev/js/editor/components/dynamic-tags/control-behavior.test.js:104]:
```suggestion
			description: 'Should not render tools when there is no Pro and there is default',
```

**Review by matipojo — DISMISSED** [review]

**Pull request merged**